uxarray/grid/geometry.py

```diff
@@ -242,3 +249,64 @@ def _grid_to_matplotlib_linecollection(grid):
 
     # need transform? consider adding it later if needed
     return LineCollection(lines)
+
+
+def _is_pole_point_inside_polygon(pole, face_edge_cart):
+    """Determines if a pole point is inside a polygon. For notice: if the pole
+    point is on the edge of the polygon, our function will consider it as
+    "inside the polygon". To use this function, the given face cannot reach out
+    the other hemisphere. For example, if you want to check if the North pole
+    is inside a polygon, the polygon should not reach the south hemisphere.
+
+    Parameters
+    ----------
+    pole : str
+        Either 'North' or 'South'
+    face_edge_cart : np.ndarray
+        Face polygon that is represented by a list of edges in Cartesian coordinates.
+        The shape of the array is (n_edges, 2, 3)
+
+    Returns
+    -------
+    bool
+        True if pole point is inside polygon, False otherwise.
+
+    Raises
+    ------
+    ValueError
+        If the provided pole point is neither 'North' nor 'South'.
+
+    Warning
+    -----
+    UserWarning
+        To use this function, the given face cannot reach out the other hemisphere. For example, if you
+        want to check if the North pole is inside a polygon, the polygon should not reach the south hemisphere.
+    """
+
+    if pole not in POLE_POINTS:
+        raise ValueError('Pole point must be either "North" or "South"')
+
+    warnings.warn(
+        'To use this function, the given face cannot reach out the other hemisphere. For example, if you '
+        'want to check if the North pole is inside a polygon, the polygon should not reach the south hemisphere.'
+    )
```

Comment:
Do we need to raise this warning everything we call this function? I think having a blurb in the docstring should suffice.

Reply:
Good point, I thought about that before. But I am not sure what the users will prefer for the handling so I just adopt a very conservative approach here.

But I am working on the algo to deal with the equator-crossing case now and hope it works out!